### ISSUE TITLE
Block user from accessing exercise contribution page

### ISSUE THREAD

**Issue by rolandgeider** [wger-project/flutter#258]
If a user has not confirmed their email (user provider, Profile), they should't be able to access the page to contribute exercises (AddExerciseScreen, Workout tab -> three dot menu). Instead, we should show a simple info text and a button that opens the user pofile form (UserProfileForm) where it is possible to confirm it

**Comment by Hirdey-1999** [comment]
Can i work on this issue as i have did the same in my app.

**Comment by rolandgeider** [comment]
Hi! Sure, it's yours

**Comment by Hirdey-1999** [comment]
Thanks for assigning

**Comment by Arnav2001** [comment]
can I work on this issue, I can resolve it.

**Comment by rolandgeider** [comment]
Hey! @Hirdey-1999 is already working on this

**Comment by Arnav2001** [comment]
do you have anything else for me

**Comment by rolandgeider** [comment]
@Arnav2001 perhaps #254 ?

**Comment by Harshit933** [comment]
Hey can you please assign me on this issue?

**Comment by rolandgeider** [comment]
Hi @Harshit933 , sure!

If you need help setting up a backend or want to use the test server, just ping me

**Comment by Harshit933** [comment]
@rolandgeider when I am running pub get i am getting this error, error is thrown from the l10n folder

![Screenshot from 2023-01-27 19-14-37](https://user-images.githubusercontent.com/90508384/215102840-7375faef-def6-49b6-969c-f356206fdf70.png)

I am quite new in open source any help would be much appreciated.

**Comment by rolandgeider** [comment]
it seems there's something wrong with the translation files, let me take a look

**Comment by Harshit933** [comment]
Okay, let me know whether should I file this as an issue.

**Comment by rolandgeider** [comment]
It seems the placeholders in the translations were translated as well, if you pull from master it should work

**Comment by rolandgeider** [comment]
BTW, you can use this server, just click on custom server in the login screen

https://wger-master.rge.uber.space/
username: user
password: flutteruser

(this is reset every 24h so feel free to change things)

**Comment by Harshit933** [comment]
@rolandgeider There are still some issues. 
![Screenshot from 2023-01-27 21-32-00](https://user-images.githubusercontent.com/90508384/215133837-89be98f0-568f-46e6-bd3e-251eb96b520d.png)

**Comment by rolandgeider** [comment]
fixed, now the app starts 😇 

Just had corrected those three errors from the first screenshot

**Comment by Harshit933** [comment]
@rolandgeider created my first PR. Review it please.

**Comment by Harshit933** [comment]
Oh I forgot to run test cases on my local system. Can you please help me with the errors i am getting on my local system  


![Screenshot from 2023-01-29 18-31-13](https://user-images.githubusercontent.com/90508384/215327860-9907d08d-7fc2-449f-bb90-c0e690ce09b2.png)

**Comment by rolandgeider** [comment]
that's a bit strange, in my machine only the only test that fails is the same as here on actions with the local preferences and which I still need to fix. Are you using flutter 3.7?

**Comment by Harshit933** [comment]
yes, i've recently updated to 3.7

**Comment by rolandgeider** [comment]
(I just pushed a commit that fixed the failing local preferences test) But I see now that I also get a warning a bout Tapping on a widget, the tests pass though, so everyting should be fine

**Comment by Harshit933** [comment]
Still having the same issue. App is not running on my device. Should I downgrade my flutter version?

**Comment by rolandgeider** [comment]
No, I'm using 3.7 as well

**Comment by Harshit933** [comment]
@rolandgeider seems like I was using a beta version of flutter 3.7, though I have fixed the issue. This is the UI which I have built let me know if any changes needs to be done (also it passes all the test cases). One more thing is that should I make a new pull request or is there any other option to modify the previous PR that i've made.


![WhatsApp Image 2023-01-30 at 6 54 19 PM](https://user-images.githubusercontent.com/90508384/215491216-98e375da-38b4-4f61-ba00-eb57be5a8e4b.jpeg)


![Screenshot from 2023-01-30 18-54-26](https://user-images.githubusercontent.com/90508384/215491242-ab196fa9-723a-473f-88db-3bfda48e1f3a.png)
